code/game/machinery/hivebot_swarm_core.dm

```diff
@@ -0,0 +1,111 @@
+/obj/machinery/hivebot_swarm_core //A hulking machine armed with several weapons. Creates hivebots and defends itself with a buzzsaw and mounted weapons.
+	name = "hivebot swarm core"
+	desc = "An incredibly dangerous machine with random weapons haphazardly attached with spot welds. Its main body seems impervious to damage, but the head is vulnerable."
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "hivebot_swarm_core"
+	pixel_x = -32
+	pixel_y = -16
+	use_power = FALSE
+	obj_integrity = 250
+	max_integrity = 250
+	anchored = TRUE
+	density = TRUE
+	var/recharge_period = 0 //The number of ticks the core is waiting for and cannot do anything
+	var/mob/living/threat_to_swarm //The core's "target" for its attacks
+	var/hivebot_limit = 50 //Stop making hivebots if there are this many in a 7x7 square around us
```

Comment:
`range = 7` means 15x15. You had me worried here: 7x7 is 49 so a limit of 50 in that box would mean more than one hivebot per turf.